servers/rendering/renderer_rd/effects/tone_mapper.cpp

```diff
@@ -182,43 +213,141 @@ void ToneMapper::tonemapper(RID p_source_color, RID p_dst_framebuffer, const Ton
+
+	RD::Uniform u_glow_map;
+	u_glow_map.uniform_type = RD::UNIFORM_TYPE_SAMPLER_WITH_TEXTURE;
+	u_glow_map.binding = 1;
+	u_glow_map.append_id(default_mipmap_sampler);
+	u_glow_map.append_id(p_settings.glow_map);
+
+	RD::Uniform u_color_correction_texture;
+	u_color_correction_texture.uniform_type = RD::UNIFORM_TYPE_SAMPLER_WITH_TEXTURE;
+	u_color_correction_texture.binding = 0;
+	u_color_correction_texture.append_id(default_sampler);
+	u_color_correction_texture.append_id(p_settings.color_correction_texture);
+
+	RID shader = tonemap_mobile.shader.version_get_shader(tonemap_mobile.shader_version, mode);
+	ERR_FAIL_COND(shader.is_null());
+
+	RD::DrawListID draw_list = RD::get_singleton()->draw_list_begin(p_dst_framebuffer);
+	RD::get_singleton()->draw_list_bind_render_pipeline(draw_list, tonemap_mobile.pipelines[mode].get_render_pipeline(RD::INVALID_ID, RD::get_singleton()->framebuffer_get_format(p_dst_framebuffer), false, RD::get_singleton()->draw_list_get_current_pass(), spec_constant));
+	RD::get_singleton()->draw_list_bind_uniform_set(draw_list, uniform_set_cache->get_cache(shader, 0, u_source_color), 0);
+	RD::get_singleton()->draw_list_bind_uniform_set(draw_list, uniform_set_cache->get_cache(shader, 1, u_glow_texture, u_glow_map), 1);
+	RD::get_singleton()->draw_list_bind_uniform_set(draw_list, uniform_set_cache->get_cache(shader, 2, u_color_correction_texture), 2);
+
+	RD::get_singleton()->draw_list_set_push_constant(draw_list, &tonemap_mobile.push_constant, sizeof(TonemapPushConstantMobile));
+	RD::get_singleton()->draw_list_draw(draw_list, false, 1u, 3u);
+	RD::get_singleton()->draw_list_end();
+}
+
+void ToneMapper::tonemapper_subpass(RD::DrawListID p_subpass_draw_list, RID p_source_color, RD::FramebufferFormatID p_dst_format_id, const TonemapSettings &p_settings) {
+	UniformSetCacheRD *uniform_set_cache = UniformSetCacheRD::get_singleton();
+	ERR_FAIL_NULL(uniform_set_cache);
+	MaterialStorage *material_storage = MaterialStorage::get_singleton();
+	ERR_FAIL_NULL(material_storage);
+
+	ERR_FAIL_COND_MSG(p_settings.use_glow, "Glow is not supported when using subpasses.");
+
+	memset(&tonemap_mobile.push_constant, 0, sizeof(TonemapPushConstantMobile));
+
+	tonemap_mobile.push_constant.bcs[0] = p_settings.brightness;
+	tonemap_mobile.push_constant.bcs[1] = p_settings.contrast;
+	tonemap_mobile.push_constant.bcs[2] = p_settings.saturation;
+
+	tonemap_mobile.push_constant.src_pixel_size[0] = 1.0 / p_settings.texture_size.x;
+	tonemap_mobile.push_constant.src_pixel_size[1] = 1.0 / p_settings.texture_size.y;
+	tonemap_mobile.push_constant.glow_intensity = p_settings.glow_intensity;
+	tonemap_mobile.push_constant.glow_map_strength = p_settings.glow_map_strength;
+
+	tonemap_mobile.push_constant.exposure = p_settings.exposure;
+	tonemap_mobile.push_constant.white = p_settings.white;
+	tonemap_mobile.push_constant.luminance_multiplier = p_settings.luminance_multiplier;
+
+	uint32_t spec_constant = 0;
```

Comment:
Nitpick, but as discussed on RC, I think we should comment out or disable these seeing they are not supported when a subpass is used. They are hardcoded to false in `RendererSceneRenderRD::_post_process_subpass` but for future people looking at this, it may be clearer to see here that these settings are only supported with a full pass.

Reply:
(well comment out the lines from the code block starting at line 328, dang preview positioning :P